strawberryfields/ops.py

```diff
@@ -1698,7 +1698,7 @@ def _decompose(self, reg):
                 if np.abs(s) >= _decomposition_tol:
                     cmds.append(Command(Sgate(s), reg[n]))
 
-            if np.all(np.abs(self.U - np.identity(len(self.U))) >= _decomposition_tol):
+            if not np.allclose(self.U, np.identity(len(self.U)), atol=_decomposition_tol):
```

Comment:
Should we add `rtol=0` here as well, so that it is an identical check to the previous version?